public/app/features/dashboard/state/PanelModel.ts

```diff
@@ -475,8 +475,12 @@ export class PanelModel implements DataConfigSource, IPanelModel {
     wasAngular: boolean
   ) {
     if (newPlugin.onPanelTypeChanged) {
+      const { targets, transformations } = this;
       const prevOptions = wasAngular ? { angular: oldOptions } : oldOptions.options;
       Object.assign(this.options, newPlugin.onPanelTypeChanged(this, oldPluginId, prevOptions, this.fieldConfig));
+      if (targets !== this.targets || transformations !== this.transformations) {
+        this.refresh(); // ???? TODO ??? is this necessary on the first one?
```

Comment:
can all changes to this file be removed?

Reply:
Agree on this - this file has a large surface area impact 😬

Reply:
remove changes to this file and the two other minor nits and I think this is good to go!